packages/gestalt/src/PageHeaderComponents.js

```diff
@@ -0,0 +1,212 @@
+// @flow strict
+import { useState, useRef, Fragment, type Element, type Node } from 'react';
+import Box from './Box.js';
+import Heading from './Heading.js';
+import Badge from './Badge.js';
+import Tooltip from './Tooltip.js';
+import IconButton from './IconButton.js';
+import Text from './Text.js';
+import Link from './Link.js';
+import Flex from './Flex.js';
+import Mask from './Mask.js';
+import Image from './Image.js';
+import Dropdown from './Dropdown.js';
+import { type ActionType } from './PageHeader.js';
+
+export function PageHeaderTitle({
+  hasThumbnail,
+  title,
+}: {|
+  hasThumbnail: boolean,
+  title: string,
+|}): Node {
+  return (
+    <Fragment>
+      <Box display="block" smDisplay="none">
+        <Heading size="400" lineClamp={1} accessibilityLevel={1}>
+          {title}
+        </Heading>
+      </Box>
+      <Box
+        display="none"
+        smDisplay="block"
+        dangerouslySetInlineStyle={{ __style: hasThumbnail ? { marginTop: '-5px' } : {} }}
+      >
+        <Heading size="500" lineClamp={1} accessibilityLevel={1}>
+          {title}
+        </Heading>
+      </Box>
+    </Fragment>
+  );
+}
+
+export function PageHeaderThumbnail({ thumbnail }: {| thumbnail: Element<typeof Image> |}): Node {
+  return (
+    <Box display="none" smDisplay="block" aria-hidden>
+      <Mask height={48} rounding={2} width={48}>
+        {thumbnail}
+      </Mask>
+    </Box>
+  );
+}
+
+export function PageHeaderBadge({
+  badgeText,
+  badgeTooltipText,
+}: {|
+  badgeText: string,
+  badgeTooltipText?: string,
+|}): Node {
+  return (
+    <Fragment>
+      {badgeText && !badgeTooltipText ? (
+        <Badge text={badgeText} type="info" position="top" />
+      ) : null}
+      {badgeText && badgeTooltipText ? (
+        <Tooltip accessibilityLabel="" text={badgeTooltipText} idealDirection="up">
+          <Badge text={badgeText} type="info" position="top" />
+        </Tooltip>
+      ) : null}
+    </Fragment>
+  );
+}
+
+export function PageHeaderHelperIconButton({
+  accessibilityLabel,
+  accessibilityControls,
+  accessibilityExpanded,
+  onClick,
+}: {|
+  accessibilityLabel: string,
+  accessibilityControls: string,
+  accessibilityExpanded: boolean,
+  onClick: ({|
+    event:
+      | SyntheticMouseEvent<HTMLButtonElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  |}) => void,
+|}): Node {
+  return (
+    <Box dangerouslySetInlineStyle={{ __style: { marginTop: '-1px' } }} rounding="circle">
```

Comment:
Applying negative margins on the outermost element of a sub-component is a bit of an anti-pattern…consider applying this where the sub-component is used instead.

Reply:
DONE